File: app/src/main/java/me/imzack/app/end/injector/component/TypeSearchComponent.java
package me.imzack.app.end.injector.component;

import me.imzack.app.end.injector.module.TypeSearchPresenterModule;
import me.imzack.app.end.injector.scope.ActivityScope;
import me.imzack.app.end.view.activity.TypeSearchActivity;

import dagger.Component;

@ActivityScope
@Component(dependencies = AppComponent.class, modules = TypeSearchPresenterModule.class)
public interface TypeSearchComponent {

    void inject(TypeSearchActivity typeSearchActivity);
}

File: app/src/main/kotlin/me/imzack/app/end/injector/component/TypeSearchComponent.kt
package me.imzack.app.end.injector.component

import dagger.Component
import me.imzack.app.end.injector.module.TypeSearchPresenterModule
import me.imzack.app.end.injector.scope.ActivityScope
import me.imzack.app.end.view.activity.TypeSearchActivity

@ActivityScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(TypeSearchPresenterModule::class))
interface TypeSearchComponent {

    fun inject(typeSearchActivity: TypeSearchActivity)
}
